src/Utilities/FlowAnalysis/Framework/DataFlow/AnalysisEntityMapAbstractDomain.cs

```diff
@@ -19,10 +19,23 @@ protected AnalysisEntityMapAbstractDomain(AbstractValueDomain<TValue> valueDomai
         protected abstract TValue GetDefaultValue(AnalysisEntity analysisEntity);
         protected abstract bool CanSkipNewEntry(AnalysisEntity analysisEntity, TValue value);
 
+        protected abstract void AssertValidEntryForMergedMap(AnalysisEntity analysisEntity, TValue value);
+        protected virtual void AssertValidAnalysisData(IDictionary<AnalysisEntity, TValue> map)
```

Comment:
Just wondering, why not `[Conditional("DEBUG")]` here?

Reply:
Compiler would not allow me to apply conditional attribute on an override.